client/app/bundles/course/assessment/submission/components/VoiceResponseAnswer.jsx

```diff
@@ -0,0 +1,228 @@
+    recordingComponentId: PropTypes.string.isRequired,
+    recording: PropTypes.bool.isRequired,
+    intl: intlShape.isRequired,
+  }
+
+  componentDidMount() {
+    const { dispatch } = this.props;
+    dispatch(recorderComponentMount());
+  }
+
+  componentWillUnmount() {
+    const { dispatch } = this.props;
+    dispatch(recorderComponentUnmount());
+  }
+
+  onStartRecord = () => {
+    const { dispatch } = this.props;
+    const recordingComponentId = this.currentRecordingComponentId();
+    recorderHelper.startRecord().then(() => dispatch(setRecording({ recordingComponentId })));
+  }
+
+  onStopRecord = field => () => {
+    const { dispatch } = this.props;
+    recorderHelper.stopRecord().then((file) => {
+      const { input: { onChange, value = {} } } = field;
+      const { url, name } = value;
+      /**
+       * check SingleFileInput about the format of the single file
+       */
+      onChange({ file, url, name });
+      return dispatch((setNotRecording()));
+    });
+  }
+
+  /**
+   * It is just a unique Id for each component make use of recorder
+   */
+  currentRecordingComponentId = () => {
+    const { question } = this.props;
+    return `voice_response_${question.id}`;
+  }
+
+  isDropFileSupported = (file) => {
+    let found = false;
+    sharedConstants.SUPPORTED_VOICE_FILE_TYPES.forEach((fileType) => {
+      if (file.type === fileType) {
+        found = true;
+      }
+    });
+    return found;
+  }
+
+  renderSingleFileInputChildren = (field) => {
+    const { input } = field;
+    const { value = {} } = input;
+    const { file = {} } = value;
+    const { name: fileName } = file;
+    return (
+      <div style={styles.singleFileInputChildrenWrapper}>
+        <div style={styles.singleFileInputChildren}>
+          <div>
+            <FormattedMessage {...translations.chooseVoiceFileExplain} />
+          </div>
+          {fileName}
+        </div>
+      </div>
+    );
+  }
+
+  renderAudio = (field) => {
+    const { input: { value } } = field;
+    const { file, url } = value;
+    let finalUrl;
+    if (file) {
+      finalUrl = URL.createObjectURL(file);
+    } else if (url) {
+      finalUrl = url;
+    }
+    if (finalUrl) {
+      return (<audio
+        controls
+        src={finalUrl}
+      >
+        <track kind="captions" />
+      </audio>);
+    }
+    return null;
+  }
+
+  renderAudioInput = (readOnly, recording, recordingComponentId, field) => {
+    if (readOnly) {
+      return null;
+    }
+    const { intl } = this.props;
+    return (
+      <div>
+        <div style={styles.fileInputWrapper}>
+          <SingleFileInput
+            disabled={readOnly}
+            isDropFileSupported={this.isDropFileSupported}
+            {...field}
+          >
+            {this.renderSingleFileInputChildren(field)}
+          </SingleFileInput>
+        </div>
+        <div>
+          <FlatButton
+            primary
+            className="btn btn-primary"
```

Comment:
These are bootstrap classes, we don't need them for React side as the styling is done by material-ui

Reply:
Will remove this